### DISCUSSION TITLE
Removal of `type` field for community objects

### DISCUSSION THREAD

**Discussion by gmazoyer** [peering-manager/peering-manager#577] [Polls]
When communities have been introduced, a `type` field was included to separate communities that may be applied when receiving or advertising a route. I believe that this field is not really useful and could be removed. Let us know what you think with this poll.

**Comment by maznu** [comment]
I found the "type" confusing, because many of the communities we import are also exported to downstreams and peers.

e.g. are these communities that `$my_asn` sets on import, or ones that e.g. `$downstream_customer` is going to set on their announcements to `$my_asn` to affect how we propagate it?

So my vote is remove the import/export `type` field, and let operators invent their own taxonomy as required (e.g. tags)

**Comment by jamesditrapani** [comment]
I currently use the `type` field to apply ingress/egress communities to routing policies:

```jinja2
{%-for community in ixp.communities.all() %}
      {%- if community.type == "ingress" %}
         set community {{ community.value }} additive
      {%-endif %}
 {%-endfor%}
  ```
  
  This is something I could probably use tags for but having the feature is nice.

**Comment by wtremmel** [comment]
Similar here - but I build  community-sets instead (don't ask about the 9999 at the end - I was to lazy to code something to omit the ","

```jinja2
! Communities for AS{{as.asn}}
large-community-set large-communities-as{{ as.asn }}-in
  {%- for community in as.communities.all() %}
    {%- if community.type == "ingress" %}
      {%- if community | has_tag("large-community") %}
 {{ community.value }},
      {%- endif %}
    {%- endif %}
  {%- endfor %}
 {{local_as.asn}}:9999:9999
end-set
large-community-set large-communities-as{{ as.asn }}-out
  {%- for community in as.communities.all() %}
    {%- if community.type == "egress" %}
      {%- if community | has_tag("large-community") %}
 {{ community.value }},
      {%- endif %}
	  {%- endif %}
  {%- endfor %}
 {{local_as.asn}}:9999:9999
end-set
```

I already use a tag to check if a community is large, so another one would be even less "elegant"...

**Comment by gmazoyer** [comment]
What if we make this field optional then?

**Reply by jamesditrapani** [reply]
> What if we make this field optional then?

I'd be happy with that 👍
